### PR TITLE
SDK-193: Automatically redirect Tomcat to OpenMRS Context

### PR DESCRIPTION
**Pull request by ODORA0**
…p://localhost:port/openmrs

**Issue worked on**: https://issues.openmrs.org/browse/SDK-193

### CODE REVIEW

**Comment by dkayiwa** [comment]
Have you tried to run this change and see what happens?

**Comment by ODORA0** [comment]
@dkayiwa Yes indeed, When I released a new version of the SDK and created a server, I was redirected to openmrs from the root context. I also tried this over `Reference Application 2.10.0`

**Comment by ODORA0** [comment]
Attached screenshots and server logs on the JIRA issue

**Comment by dkayiwa** [comment]
What are the exact steps that i should do to test your fix?

**Comment by ODORA0** [comment]
- Release a new version of the SDK according to your SDK version with the command `mvn org.openmrs.maven.plugins:openmrs-sdk-maven-plugin:3.13.5-SNAPSHOT:setup`
- Run `mvn openmrs-sdk:run` to start the newly created server
- Access openmrs through the browser according to the port its running on (http://localhost:8082/openmrs)

It redirects to OpenMRS from the root context without any clicks made.

**Comment by dkayiwa** [comment]
@ssmusoke that does not look like what @ssmusoke was asking for. Can you first cross check with him and ask if this is the behaviour he expected?

**Comment by dkayiwa** [comment]
@ODORA0 or are you looking for the same functionality and what happens in the standalone version of openmrs where the browser is automatically launched with the appropriate url? https://github.com/openmrs/openmrs-standalone/blob/master/src/main/java/org/openmrs/standalone/StandaloneUtil.java#L256-L275

**Comment by ssmusoke** [comment]
@ODORA0 This does not work for me as I get nothing when I run http://localhost:port although http://localhost:port/openmrs works

**Comment by ssmusoke** [comment]
> @ODORA0 or are you looking for the same functionality as what happens in the standalone version of openmrs where the browser is automatically launched with the appropriate url? https://github.com/openmrs/openmrs-standalone/blob/master/src/main/java/org/openmrs/standalone/StandaloneUtil.java#L256-L275

@dkayiwa @ODORA0 I am not looking at automatically opening the browser, rather just being able to use http://localhost:port to automatically redirect to http://localhost:port/openmrs (as usual)

**Comment by ODORA0** [comment]
@dkayiwa @ssmusoke Please review this. So I created a servlet to do a proper redirect and add that to the root context and it works on my machine. So to test this fix

1. Run the command `mvn org.openmrs.maven.plugins:openmrs-sdk-maven-plugin:3.13.5-SNAPSHOT:run`
2. Navigate to http://localhost:port, this will automatically redirect you to http://localhost:port/openmrs on the browser

File: integration-tests/src/test/resources/integration-test/buildIT/module2/api/src/main/resources/moduleApplicationContext.xml [hunk 1]
```diff
@@ -18,6 +18,7 @@
   		    http://www.springframework.org/schema/util/spring-util-3.0.xsd">
 
 	<!-- Add here beans related to the API context -->
+	<Context path="/" docBase="${catalina.home}/webapps/openmrs"/>
```
Comment (ssmusoke) [inline integration-tests/src/test/resources/integration-test/buildIT/module2/api/src/main/resources/moduleApplicationContext.xml:21]: @ODORA0 do we still need this here with the Servlet approach?

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/OpenmrsRedirectionServlet.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.catalina.servlets.DefaultServlet;
+public class OpenmrsRedirectionServlet extends DefaultServlet {
```
Comment (ssmusoke) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/OpenmrsRedirectionServlet.java:13]: @ODORA0 rename to OpenMRSSDKRedirectServlet

Comment (ssmusoke) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/OpenmrsRedirectionServlet.java:13]: @ODORA0 Add a class level comment to describe what this Servlet does

**Comment by ODORA0** [comment]
@ssmusoke @dkayiwa Please review, added a few test

**Comment by dkayiwa** [comment]
Have you tested and confirmed that it is working as expected?

**Comment by ODORA0** [comment]
@dkayiwa yes I have tested the fix and it works

**Comment by dkayiwa** [comment]
How does it behave?

**Comment by ODORA0** [comment]
1. Run the command `mvn org.openmrs.maven.plugins:openmrs-sdk-maven-plugin:3.13.5-SNAPSHOT:run`
2. Navigate to http://localhost:port, this will automatically redirect you to http://localhost:port/openmrs on the browser

**Comment by ssmusoke** [comment]
@dkayiwa This PR is good to go, so can be merged

**Comment by dkayiwa** [comment]
@ssmusoke with your dev stage, feel free to merge pull requests that you feel are ripe. 👍

**Pull request merged**